test/test.py

```diff
@@ -34,10 +34,8 @@ def stdout_redirect():
     else:
         output = io.BytesIO()
     sys.stdout = output
-    try:
-        yield output
-    finally:
-        output.close()
+
+    yield output
```

Comment:
Why remove the `finally` logic? This code doesn't hide exceptions, it just stops them screwing up the IO. (At least I think that's what it does, it has been a while!)

Reply:
I removed this because it was causing a test failure for me, where the first test calling `stdout_redirect()` ran fine but the second would throw `ValueError: I/O operation on closed file`

<img width="1157" alt="image" src="https://user-images.githubusercontent.com/51478689/221667171-4b7b85fa-0f6c-4cc1-8f93-2fcb451421c5.png">

Reply:
I'm not having much luck reproducing this, but I see the problem. This error shouldn't happen the next time stdout_redirect is used: it will happen the next time something writes to STDOUT _without_ the redirect, because the decorator leaves it pointing to the closed file. How about this?

```python
@contextmanager
def stdout_redirect():
    """Enable tests to inspect stdout in suitable format for Python version"""
    if sys.version_info > (3,):
        output = io.StringIO()
    else:
        output = io.BytesIO()
    sys_stdout = sys.stdout
    sys.stdout = output

    try:
        yield output
    finally:
        output.close()
        sys.stdout = sys_stdout
```

Reply:
Yep just checked and it passes all tests fine on my local! Have committed now ✅